pkg/cli/pod/pod.go

```diff
@@ -0,0 +1,398 @@
+	if err != nil {
+		return err
+	}
+
+	client := kubeclientset.NewForConfigOrDie(config)
+	pods, err := client.CoreV1().Pods(listPodFlags.Namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: labelSelector.String(),
+	})
+	if err != nil {
+		return err
+	}
+
+	if len(pods.Items) == 0 {
+		fmt.Printf("No resources found\n")
+		return nil
+	}
+	PrintPods(pods, os.Stdout)
+
+	return nil
+}
+
+func PrintPods(pods *corev1.PodList, writer io.Writer) {
+	maxNameLen := 0
+	maxReadyLen := 0
+	maxStatusLen := 0
+	maxRestartLen := 0
+	maxAgeLen := 0
+
+	var infoList []PodInfo
+	for _, pod := range pods.Items {
+		info := printPod(&pod)
+		infoList = append(infoList, info)
+		// update max length for each column
+		if len(info.Name) > maxNameLen {
+			maxNameLen = len(info.Name)
+		}
+		if len(info.ReadyContainers) > maxReadyLen {
+			maxReadyLen = len(info.ReadyContainers)
+		}
+		if len(info.Status) > maxStatusLen {
+			maxStatusLen = len(info.Status)
+		}
+		if len(info.Restarts) > maxRestartLen {
+			maxRestartLen = len(info.Restarts)
+		}
+		if len(info.CreationTimestamp) > maxAgeLen {
+			maxAgeLen = len(info.CreationTimestamp)
+		}
+	}
+	columnSpacing := 8
+	maxNameLen += columnSpacing
+	maxReadyLen += columnSpacing
+	maxStatusLen += columnSpacing
+	maxRestartLen += columnSpacing
+	maxAgeLen += columnSpacing
+	formatStr := fmt.Sprintf("%%-%ds%%-%ds%%-%ds%%-%ds%%-%ds\n", maxNameLen, maxReadyLen, maxStatusLen, maxRestartLen, maxAgeLen)
+	_, err := fmt.Fprintf(writer, formatStr, Name, Ready, Status, Restart, Age)
+	if err != nil {
+		fmt.Printf("Failed to print Pod information: %s.\n", err)
+		return
+	}
+	for _, info := range infoList {
+		_, err := fmt.Fprintf(writer, formatStr, info.Name, info.ReadyContainers, info.Status, info.Restarts, info.CreationTimestamp)
+		if err != nil {
+			fmt.Printf("Failed to print Pod information: %s.\n", err)
+			return
+		}
+	}
+}
+
+// createLabelSelector creates a label selector based on the provided job name or queue name.
+func createLabelSelector(jobName, queueName string) (labels.Selector, error) {
+	var labelSelector labels.Selector
+	reqs := []labels.Requirement{}
+
+	// Add label requirements based on jobName label
+	if jobName != "" {
+		inRequirement, err := labels.NewRequirement(v1alpha1.JobNameKey, selection.In, []string{jobName})
+		if err != nil {
+			return nil, err
+		}
+		reqs = append(reqs, *inRequirement)
+	}
+	// Add label requirements based on queueName label
+	if queueName != "" {
+		// pods under vcjobs
+		inRequirementForVcJob, err := labels.NewRequirement(v1alpha1.QueueNameKey, selection.In, []string{queueName})
+		if err != nil {
+			return nil, err
+		}
+		// pods under specific queue
+		inRequirementForPod, err := labels.NewRequirement(schedulingv1beta1.QueueNameAnnotationKey, selection.In, []string{queueName})
+		if err != nil {
+			return nil, err
+		}
+		reqs = append(reqs, *inRequirementForVcJob)
+		reqs = append(reqs, *inRequirementForPod)
```

Comment:
If I'm not wrong, the label selectors is an AND operation and all labels.Requirements should be match when use label.Match method, so when user specified both job name and queue name, `inRequirementForPod` label selector will also take effect and will return an empty list finally.

So you should also add a test case that both vcjob name and queue name are specified, and seperate filter selector for bare pod and vcjob pod if I'm not wrong.

Reply:
I'll test it, thanks for the reminder.